nvim/child_process_stream.lua

```diff
@@ -19,7 +19,9 @@ function ChildProcessStream.spawn(argv, env)
     stdio = {self._child_stdin, self._child_stdout, 2},
     args = args,
     env = env,
-  }, function()
+  }, function(code, signal)
+    self.exitcode = code
+    self.exitsignal = signal
     self:close()
```

Comment:
Might be good to have this as a module function, so it could be changed / and/or pass in a callback additionally somehow.
Then Neovim could use `busted.fail` here directly maybe.